### PR TITLE
Sync main from dev

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

$subject

As the GraalVM check is passing for the branch, going ahead with the synching. [https://github.com/ballerina-platform/module-ballerina-persist/actions/runs/4988833284/jobs/8948164966](https://github.com/ballerina-platform/module-ballerina-persist/actions/runs/4988833284/jobs/8948164966)

related issues : [4451](https://github.com/ballerina-platform/ballerina-standard-library/issues/4451)

## Checklist
- [ ] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/stream_types.bal [hunk 1]
```diff
@@ -78,6 +78,70 @@ public class PersistSQLStream {
     }
 }
 
+
+public class PersistGoogleSheetsStream {
```
Comment (daneshk) [inline ballerina/stream_types.bal:82]:
```suggestion
public class GoogleSheetsStream {
```

Comment (daneshk) [reply ballerina/stream_types.bal:82]: we don't need Persist prefix here

File: ballerina/metadata_types.bal [hunk 2]
```diff
@@ -52,11 +66,43 @@ public type EntityFieldMetadata record {|
     RelationMetadata relation;
 |};
 
+# Represents the metadata associated with a field of an entity.
+# Only used by the generated persist clients and `persist:SQLClient`.
```
Comment (daneshk) [inline ballerina/metadata_types.bal:70]:
```suggestion
# Only used by the generated persist clients and `persist:GoogleSheetsClient`.
```

File: ballerina/stream_types.bal (hunk 1 above)
Comment (sahanHe) [reply ballerina/stream_types.bal:82]: We use the same convention for `PersistSQLStream`

File: ballerina/metadata_types.bal (hunk 2 above)
Comment (sahanHe) [reply ballerina/metadata_types.bal]: updated in [0639ae4](https://github.com/ballerina-platform/module-ballerina-persist/pull/211/commits/0639ae4b7a33d2a6f8534c4823231425175dea94)

**Review by daneshk — APPROVED** [review]

**Pull request merged**